infra/binder/src/main/java/org/apache/shardingsphere/infra/binder/context/segment/insert/values/OnConflictUpdateContext.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.shardingsphere.sql.parser.statement.core.extractor.ColumnExtractor;
+import org.apache.shardingsphere.sql.parser.statement.core.extractor.ExpressionExtractor;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.assignment.ColumnAssignmentSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.column.ColumnSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.expr.ExpressionSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.expr.FunctionSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.expr.simple.LiteralExpressionSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.expr.simple.ParameterMarkerExpressionSegment;
+import org.apache.shardingsphere.sql.parser.statement.core.segment.dml.predicate.WhereSegment;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * On conflict context.
+ */
+@Getter
+public final class OnConflictUpdateContext {
+    
+    private final int parameterCount;
+    
+    private final List<ExpressionSegment> valueExpressions;
+    
+    private final Collection<WhereSegment> whereSegments = new LinkedList<>();
+    
+    private final Collection<ColumnSegment> columnSegments;
+    
+    private final List<ParameterMarkerExpressionSegment> parameterMarkerExpressions;
+    
+    private final List<Object> parameters;
+    
+    public OnConflictUpdateContext(final Collection<ColumnAssignmentSegment> assignments, final List<Object> params, final int parametersOffset, final Optional<WhereSegment> segment) {
+        List<ExpressionSegment> expressionSegments = assignments.stream().map(ColumnAssignmentSegment::getValue).collect(Collectors.toList());
+        segment.ifPresent(whereSegments::add);
+        for (WhereSegment whereSegment : whereSegments) {
+            expressionSegments.add(whereSegment.getExpr());
+        }
+        columnSegments = assignments.stream().map(each -> each.getColumns().get(0)).collect(Collectors.toList());
+        ColumnExtractor.extractColumnSegments(columnSegments, whereSegments);
+        valueExpressions = getValueExpressions(expressionSegments);
+        parameterMarkerExpressions = ExpressionExtractor.getParameterMarkerExpressions(expressionSegments);
+        parameterCount = parameterMarkerExpressions.size();
+        parameters = getParameters(params, parametersOffset);
+    }
+    
+    /**
+     * get value expressions from expression segments.
+     *
+     * @param assignments Collection of expression segments
+     * @return List of value expressions
+     */
+    private List<ExpressionSegment> getValueExpressions(final Collection<ExpressionSegment> assignments) {
+        List<ExpressionSegment> result = new ArrayList<>(assignments.size());
```

Comment:
Only for cast Collection to List? If so, please modify the origianl assignments to ArrayList.